### ISSUE TITLE
`@stylistic/indent`: rule is broken with version 2.2.0

### ISSUE THREAD

**Issue by xavierfoucrier** [eslint-stylistic/eslint-stylistic#421]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Since I updated to version 2.2.0, it looks like there is an issue when using the rule `@stylistic/indent`.
Commenting the lines below make the linter work again.

```js
'@stylistic/indent': [
  'error',
  2, {
    'SwitchCase': 1,
  },
],
```

Error log:
```sh
Run npm run lint

> github-project@0.0.1 lint
> eslint src/**/*.js


Oops! Something went wrong! :(

ESLint: 9.5.0

TypeError: Cannot read properties of undefined (reading 'length')
Occurred while linting /home/runner/work/github-project/github-project/src/component.js:10
Rule: "@stylistic/indent"
    at PropertyDefinition (/home/runner/work/github-project/github-project/node_modules/@stylistic/eslint-plugin-ts/dist/configs.js:776:30)
    at ruleErrorHandler (/home/runner/work/github-project/github-project/node_modules/eslint/lib/linter/linter.js:1092:48)
    at /home/runner/work/github-project/github-project/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/runner/work/github-project/github-project/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/runner/work/github-project/github-project/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/home/runner/work/github-project/github-project/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/home/runner/work/github-project/github-project/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at runRules (/home/runner/work/github-project/github-project/node_modules/eslint/lib/linter/linter.js:1136:40)
    at Linter._verifyWithFlatConfigArrayAndWithoutProcessors (/home/runner/work/github-project/github-project/node_modules/eslint/lib/linter/linter.js:1879:31)
```

Thanks for the feedback! 😉 

> Downgrading to version 2.1.0 works great.

### Reproduction

eslint src/**/*.js

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by xavierfoucrier** [comment]
Thanks @9romise for working on this 😎💛
